gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/MaximumVersion.java

```diff
@@ -25,7 +25,7 @@
 import java.util.Optional;
 
 /**
- * A typed version of a maximum version as declared in {@link RecommendedProductDependency#maximumVersion}.
+ * A typed version of a maximum version as declared in {@link ProductDependency#getMaximumVersion}.
```

Comment:
do we need this file? can we replace with methods exposed by sls-versions-java?

Reply:
yes we do, it allows comparing two types from sls-versions-java that aren't otherwise comparable.

Reply:
as discussed offline, since we don't actually support arbitrary OrderableSlsVesion for maximumVersion, removed this class and simplified the merging logic.